hoptimator-operator/src/main/java/com/linkedin/hoptimator/operator/Operator.java

```diff
@@ -21,27 +21,36 @@
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
+import java.util.Properties;
 
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-/** Single handle to all the clients required by all the controllers. */
+/** Single handle to all the clients and configs required by all the controllers. */
 public class Operator {
   private final static Logger log = LoggerFactory.getLogger(Operator.class);
  
   private final String namespace; 
   private final ApiClient apiClient;
   private final SharedInformerFactory informerFactory;
   private final Map<String, ApiInfo<?, ?>> apiInfo = new HashMap<>();
+  private final Properties properties;
 
-  public Operator(String namespace, ApiClient apiClient, SharedInformerFactory informerFactory) {
+  public Operator(String namespace, ApiClient apiClient, SharedInformerFactory informerFactory,
+      Properties properties) {
     this.namespace = namespace;
     this.apiClient = apiClient;
     this.informerFactory = informerFactory;
+    this.properties = properties;
```

Comment:
I think it is better to avoid potentially `null` fields. Add test for input or require non-null. ie. 
```
# test
this.properties = properties == null ? new Properties() : properties;

# validate
this.properties = Validate.notNull(properties);
```